models/comments-model.js

```diff
@@ -0,0 +1,24 @@
+const db = require("../db/connection");
+
+exports.selectCommentsByArticleId = (article_id) => {
+  return db
+    .query(
+      `SELECT * FROM comments
+       WHERE article_id = $1
+       ORDER BY created_at DESC`,
+      [article_id],
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return db
+          .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+          .then(({ rows }) => {
+            if (rows.length === 0) {
+              return Promise.reject({ status: 404, msg: "Article not found" });
+            }
+            return [];
+          });
+      }
+      return rows;
+    });
```

Comment:
Try to avoid nesting .then also try to have only one db.query in your model. 
This could be refactored to use your articleById function in the controller and chain a .then() e.g.

articleById(id).then(() => { selectCommentsByArticleId etc.

If the article doesn't exist the first function will throw an err that will get caught in the catch, if it is found the selectComments function will run